k8s/k8s.go

```diff
@@ -158,21 +161,35 @@ type PersistentVolumeClaimOps interface {
 	DeletePersistentVolumeClaim(*v1.PersistentVolumeClaim) error
 	// ValidatePersistentVolumeClaim validates the given pvc
 	ValidatePersistentVolumeClaim(*v1.PersistentVolumeClaim) error
-	// GetVolumeForPersistentVolumeClaim returns the back volume for the given PVC
+	// GetVolumeForPersistentVolumeClaim returns the volumeID for the given PVC
 	GetVolumeForPersistentVolumeClaim(*v1.PersistentVolumeClaim) (string, error)
 	// GetPersistentVolumeClaimParams fetches custom parameters for the given PVC
 	GetPersistentVolumeClaimParams(*v1.PersistentVolumeClaim) (map[string]string, error)
 	// GetPersistentVolumeClaimStatus returns the status of the given pvc
 	GetPersistentVolumeClaimStatus(*v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaimStatus, error)
 }
 
+type SnapshotOps interface {
+	// CreateSnapshot creates the given snapshot
+	CreateSnapshot(*snap_v1.VolumeSnapshot) (*snap_v1.VolumeSnapshot, error)
+	// DeletePersistentVolumeClaim deletes the given snapshot
+	DeleteSnapshot(*snap_v1.VolumeSnapshot) error
```

Comment:
I think as an interface we should be passing the name and namespace in delete calls. We did in some places but it is pending in other places. 
The advantage is when we don't have the object, but given just the name and namespace you can still delete it. We don't have to get the object and then pass it to the delete. What do you think?

Reply:
Yes, definitely better to just use name and namespace. I copied the interface from volumes for these. I'll update it